code/modules/mob/living/carbon/human/update_icons.dm

```diff
@@ -710,13 +748,24 @@ Applied by gun suicide and high impact bullet executions, removed by rejuvenate,
 	if(!on_fire)
 		set_light_on(FALSE)
 		return
+	var/fire_sprite_sheet
+	var/fire_sprite_prefix
+	if (ismonkey(src)) //Point towards burning sprites for monkeys
+		fire_sprite_sheet = 'icons/mob/humans/onmob/OnFire.dmi'
+		fire_sprite_prefix = "monkey"
+	else if (isk9synth(src)) //Point towards burning sprites for k9_synths
+		fire_sprite_sheet = 'icons/mob/humans/onmob/OnFire.dmi'
+		fire_sprite_prefix = "k9"
+	else //Default to human shaped fire sprites
+		fire_sprite_sheet = 'icons/mob/humans/onmob/OnFire.dmi'
+		fire_sprite_prefix = "Standing"
```

Comment:
we should never be type checking src, why aren't these stored on the species?

Reply:
We now declare the sheet and prefix on species.